src/groupchats.c

```diff
@@ -1672,14 +1676,20 @@ static void groupchat_onGroupSelfNickChange(ToxWindow *self, Toxic *toxic, uint3
 
     length = MIN(length, TOX_MAX_NAME_LENGTH - 1);
     memcpy(chat->peer_list[peer_index].name, new_nick, length);
-    chat->peer_list[peer_index].name[length] = 0;
+    chat->peer_list[peer_index].name[length] = '\0';
     chat->peer_list[peer_index].name_length = length;
 
     line_info_add(self, toxic->c_config, true, old_nick, chat->peer_list[peer_index].name, NAME_CHANGE, 0,
                   MAGENTA, " is now known as ");
 
     groupchat_update_last_seen(groupnumber, peer_id);
     groupchat_update_name_list(groupnumber);
+
+    ChatContext *ctx = self->chatwin;
+
+    char log_event[TOX_MAX_NAME_LENGTH * 2 + 32];
```

Comment:
No need for `TOX_MAX_NAME_LENGTH * 2` when the previous name is added in write_to_log?